src/hooks/useNodeViewPortals.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ReactPortal, useCallback, useState } from "react";
+import { createPortal } from "react-dom";
+
+import {
+  PORTAL_REGISTRY_ROOT_KEY,
+  PortalRegistryContext,
+  PortalRegistryKey,
+} from "../contexts/PortalRegistryContext.js";
+import { RegisterElement } from "../nodeViews/createReactNodeViewConstructor.js";
+
+import { useEditorState } from "./useEditorState.js";
+
+type Props = {
+  portals: Record<PortalRegistryKey, ReactPortal[]>;
+};
+
+function PortalTree({ portals }: Props) {
```

Comment:
Should we call this `PortalRegistry` in light of the context it provides being `PortalRegistryContext`?

Reply:
I think my answer is no (but there's still probably some other better name for it). Here's why:

This component actually renders the entire portal tree. The fact that there's a context and a registry at all is an implementation detail; the only components that consume the "portal registry" are descendants of this component. 

This component directly renders the "top-level" portals. Each of them checks the registry and then render their direct child node view portals, and so on and so forth until the whole tree has been rendered.

No other components ever actually need to be aware of or consume the portal registry, so I think it makes more sense to think of this component as `NodeViews` or `Portals`. `PortalTree` is maybe somewhat redundant, since all React components are trees, more-or-less, so maybe we should call it `NodeViewPortals` or something instead?

Reply:
Or even just `NodeViews`?

Reply:
Yup, I like `NodeViews`!